macro/src/main/scala/SparkSchemaHelper.scala

```diff
@@ -60,12 +60,16 @@ class SparkSchemaHelperImpl(val c: Context) {
           param.typeSignature
         }
         val isSequenceWithOptionalVal = innerType <:< optionSeq
-        val isMapWithOptionalVal = innerType <:< optionMap
         val isSequenceOfAny = innerType <:< seqAny
-        val isMapOfAny = innerType <:< mapAny
+        val isMapOfString = innerType <:< mapString
+        if (!isMapOfString && innerType.getClass == classOf[Map[_, _]]) {
+          throw new Exception(s"Only Map[String, String] is supported, not $innerType")
```

Comment:
Isn't this excluding all other types of maps? It's possible to store a map<string, long> in a RAW column, for example, and that's allowed.

Reply:
Of course, this is for deriving the schema for our own types. :facepalm: All good then.